### ISSUE TITLE
Error during `nuxi generate` in link[href] with data url containing slashes

### ISSUE THREAD

**Issue by fabon-f** [nuxt/nuxt#22134]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-nj2nvg?file=pages%2Findex.vue

### Describe the bug

Running `nuxi generate` raises an error when a project contains a HTML snippet like below.

```html
<link rel="icon" href="data:image/png;base64,aaa//bbbbbb/ccc">
```

In this reproduction, nitro recognize `/ccc` as a linked page path by mistake. Route `/ccc` doesn't exist so it causes error in `nuxi generate`.

(workaround: setting `nitro.prerender.failOnError` config to `false` to simply ignore 404 error in prerendering)

**Comment by bigshans** [comment]
Maybe the `//` caused this error.

**Comment by danielroe** [comment]
This is likely an upstream issue in nitro with the regexp for extracting link URLs.

**Comment by danielroe** [comment]
The default should be `true` as otherwise it will conceal rendering errors. (Even if it isn't a strict fix, we would have turned it on in 3.7 anyway.)

You can always add erroring URLs to `nitro.prerender.ignore`, or (less safe) disable `nitro.prerender.failOnError`.

**Comment by fabon-f** [comment]
I confirmed that the bug is resolved on Nuxt v3.7.0 thanks to the release of unjs/ufo v1.3.0. I think this issue can be closed.
